Control/Monad/Primitive.hs

```diff
@@ -225,6 +229,20 @@ instance PrimBase (L.ST s) where
   {-# INLINE internal #-}
 #endif
 
+-- | 'PrimMonad''s state token type can be annoying to handle
+--   in constraints. This typeclass lets users (visually) notice
+--   'PrimState' equality constraints less, by witnessing that
+--   @s ~ 'PrimState' m@.
+class (PrimMonad m, s ~ PrimState m) => MonadPrim s m | m -> s where
+instance (PrimMonad m, s ~ PrimState m) => MonadPrim s m
+
+-- | 'PrimBase''s state token type can be annoying to handle
+--   in constraints. This typeclass lets users (visually) notice
+--   'PrimState' equality constraints less, by witnessing that
+--   @s ~ 'PrimState' m@.
+class (PrimBase m, s ~ PrimState m) => MonadPrimBase s m | m -> s where
```

Comment:
I think it would make sense to define this class in a similar fashion that `PrimBase` is defined:
```haskell
class (PrimBase m, MonadPrim s m) => MonadPrimBase s m
````